kong/pdk/private/response/handler/utils.lua

```diff
@@ -0,0 +1,31 @@
+local pl_stringx = require "pl.stringx"
+
+local str_find = string.find
+local str_sub = string.sub
+local endswith = pl_stringx.endswith
+
+local _M = {}
+
+
+function _M.mime_type_match(this_type, this_subtype, other_type, other_subtype)
+  if this_type == other_type or this_type == "*" then
```

Comment:
Perform validation early and return as early as possible:
```
if this_type ~= other_type and this_type ~= "*" then
  return false
end
-- this_type == other_type or this_type == "*"
...
```

Reply:
There are too many false conditions. I feel do the reverse check would make the code cleaner.

```
  if condition1 then
     if condition2 then
        if condition3 then
           -- yes!
        end
     end
  end
  
  -- otherwise false
  return false
```

Reply:
I think less nesting would be cleaner and easier to modify later.